articles/building-xml-webpages-using-xslt/index.md

```diff
@@ -0,0 +1,302 @@
+      <td>
+          <xsl:value-of select="event_name" />
+      </td>
+      <td>
+          <xsl:value-of select="event_desc" />
+      </td>
+      <xsl:choose>
+          <xsl:when test="event_type_participation = 'Team'">
+              <td bgcolor="#f2f2f242">
+                  <xsl:value-of select="event_type_participation" />
+              </td>
+          </xsl:when>
+          <xsl:otherwise>
+              <td bgcolor="#cccccc">
+                  <xsl:value-of select="event_type_participation" />
+              </td>
+          </xsl:otherwise>
+      </xsl:choose>
+      <td>
+          <xsl:value-of select="event_timing/event_start_datetime" />
+      </td>
+      <td>
+          <xsl:value-of select="event_timing/event_end_datetime" />
+      </td>
+      <td>
+          <xsl:value-of select="event_organizer_email" />
+      </td>
+      <td>
+          <xsl:value-of select="event_organizer_phone" />
+      </td>
+  </tr>
+</xsl:for-each>
+```
+
+For XML tags `event_start_datetime` and `event_end_datetime`, we specify XPath as `event_timing/event_start_datetime` and `event_timing/event_end_datetime`, since the XML document contains nested XML tags.
+
+#### Additional features
+##### Counting
+Let's say, we have to count the total number of `event_id` in the XML document. We can do this by using `count()` by specifying the path for accessing the tag. For accessing `event_id` XML tag, we follow the below XML structural path:
+
+```xml
+<root>
+    <event>
+        <event_id> </event_id>
+    </event>
+</root>
+```
+
+For accessing `event_id`, the path would be `root/event/event_id`. And, to access the values, we use `xsl:value-of select=""`. By combining the above steps, we can count the number of events as shown below:
+
+```xml
+<p>
+    Number of Events - <xsl:value-of select="count(root/event/event_id)" /><br/>
+</p>
+```
+
+##### Sorting
+Now, let's say we have to sort the table rows in ascending order based on `event_id`. We have an exclusive XSLT tag for sorting. In XML, every data is considered as a `string` by default. So, we must convert it to an integer by using `number()`. `xsl:sort` helps us sort the table rows based on `number(event_id)`.
+
+```xml
+<xsl:sort select="number(event_id)" data-type="number" />
+```
+
+To try out this, replace the above code snippet with `<!--ADDITIONAL FEATURE-->` under [Full Code - Table Rows](#full-code-table-rows).
+
+#### View Output
+By following the above step-by-step procedures, you would have successfully built a webpage. To view the page, you must open the file with a `.xml` extension.
+
+Due to security concerns in several browsers like Google Chrome, the output can be viewed using the Internet Explorer browser. More details about the security concern in Google Chrome can be found [here](https://stackoverflow.com/questions/4558160/xsl-not-working-in-google-chrome).
+
+![A sample output of XML Webpage](/engineering-education/building-xml-webpages-using-xslt/output.PNG)
```

Comment:
Even on internet explorer, I fail to get the output. I see the code itself. Any alternatives?

Reply:
Hi @lalith1403. I had made a minor error in my XML document, so it didn't work. The fixed code can be found in the latest commit.

For reference, I have put links containing all the codes under "Conclusion". Please let me know if it doesn't work. Thanks.

Reply:
Yep. They worked. Thanks.